src/client/Microsoft.Identity.Client/Kerberos/Win32/SspiSecurityContext.cs

```diff
@@ -0,0 +1,195 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Runtime.InteropServices;
+
+using static Microsoft.Identity.Client.Kerberos.Win32.NativeMethods;
+
+namespace Microsoft.Identity.Client.Kerberos.Win32
+{
+#pragma warning disable 618 // This workaround required for Native Win32 API call
+
+    internal partial class SspiSecurityContext : IDisposable
+    {
+        private const int SECPKG_CRED_BOTH = 0x00000003;
+        private const int SECURITY_NETWORK_DREP = 0x00;
+
+        private const int _maxTokenSize = 16 * 1024;
+
+        private const InitContextFlag _defaultRequiredFlags =
+                                    InitContextFlag.Connection |
+                                    InitContextFlag.ReplayDetect |
+                                    InitContextFlag.SequenceDetect |
+                                    InitContextFlag.Confidentiality |
+                                    InitContextFlag.AllocateMemory |
+                                    InitContextFlag.Delegate |
+                                    InitContextFlag.InitExtendedError;
+
+        private readonly HashSet<object> _disposable = new HashSet<object>();
+
+        private readonly Credential _credential;
+        private readonly InitContextFlag _clientFlags;
+
+        private SECURITY_HANDLE _credentialsHandle;
+        private SECURITY_HANDLE _securityContext;
+        private long _logonId;
+
+        public SspiSecurityContext(
+            Credential credential,
+            string package,
+            long logonId = 0,
+            InitContextFlag clientFlags = _defaultRequiredFlags)
+        {
+            this._credential = credential;
+            this._clientFlags = clientFlags;
+            this.Package = package;
+            this._logonId = logonId;
+        }
+
+        public string Package { get; private set; }
+
+        private static void ThrowIfError(uint result)
+        {
+            if (result != 0 && result != 0x80090301)
+            {
+                throw new Win32Exception((int)result);
+            }
+        }
+
+        public ContextStatus InitializeSecurityContext(string targetName, out byte[] clientRequest)
+        {
+            var targetNameNormalized = targetName.ToLowerInvariant();
+
+            clientRequest = null;
+
+            // 1. acquire
+            // 2. initialize
+            // 3. ??

```

Comment:
What are these comments trying to say?

Reply:
This code part just copied from Kerberos.NET public package (https://www.nuget.org/packages/Kerberos.NET/) to support Windows Ticket Cache interface. I want to keep the code as it is.